src/slip.rs

```diff
@@ -87,22 +88,37 @@ impl Slip {
         self.block_hash
     }
 
-    // Set the `Block` id
+    /// Return a byte array of `Slip` data
+    pub fn hash(&self) -> [u8; 32] {
+        let mut data: Vec<u8> = vec![];
+
+        let slip_id_bytes: [u8; 8] = self.slip_id.to_be_bytes();
+        let tx_id_bytes: [u8; 8] = self.tx_id.to_be_bytes();
+        let block_id_bytes: [u8; 8] = self.block_id.to_be_bytes();
+
+        data.extend(&slip_id_bytes);
+        data.extend(&tx_id_bytes);
+        data.extend(&block_id_bytes);
+
+        hash(&data)
+    }
+
+    /// Set the `Block` id
     pub fn set_block_id(&mut self, block_id: u64) {
         self.block_id = block_id;
     }
 
-    // Set the `Transaction` id
+    /// Set the `Transaction` id
     pub fn set_tx_id(&mut self, tx_id: u64) {
         self.tx_id = tx_id;
```

Comment:
will this stay `tx` or also changed to `transaction`?

Reply:
this could change, but that refactor is outside of the scope of this PR